### PR TITLE
Clear out our projection buffers when we shutdown the pr eview window to workaround an editor bug.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide @Pilchie @kuhlenh

File: src/EditorFeatures/Core/Implementation/Preview/PreviewFactoryService.cs [hunk 1]
```diff
@@ -564,24 +566,27 @@ private ITextBuffer CreateNewPlainTextBuffer(TextDocument document, Cancellation
...
-                    leftWorkspace.Dispose();
-                    leftWorkspace = null;
-                }
-
-                if (rightWorkspace != null)
-                {
-                    rightWorkspace.Dispose();
-                    rightWorkspace = null;
-                }
+                // Workaround Editor bug.  The editor has an issue where they sometimes crash when
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Implementation/Preview/PreviewFactoryService.cs]: Should this link to the bug? Is this something you'd want reverted once the bug was fixed?

File: src/EditorFeatures/Core/Implementation/Preview/PreviewFactoryService.cs [hunk 2]
```diff
@@ -564,24 +566,27 @@ private ITextBuffer CreateNewPlainTextBuffer(TextDocument document, Cancellation
...
+                // trying to apply changes to projection buffer.  So, when the user actually invokes
+                // a SuggestedAction we may then edit a text buffer, which the editor will then 
+                // try to propagate through the projections we have here over that buffer.  To ensure
+                // that that doesn't happen, we clear out the projections first so that this crash
+                // won't happen.
+                originalBuffer.DeleteSpans(0, originalBuffer.CurrentSnapshot.SpanCount);
+                changedBuffer.DeleteSpans(0, changedBuffer.CurrentSnapshot.SpanCount);
+
+                leftWorkspace?.Dispose();
+                leftWorkspace = null;
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Implementation/Preview/PreviewFactoryService.cs]: I know this isn't your doing, but did you get any sense of why we're nulling out workspaces while you were poking around here?

File: src/EditorFeatures/Core/Implementation/Preview/PreviewFactoryService.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Preview/PreviewFactoryService.cs]: I haved pinged David for a bug-id (if they have one).

File: src/EditorFeatures/Core/Implementation/Preview/PreviewFactoryService.cs (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Preview/PreviewFactoryService.cs]: Nope.  my guess is a paranoid desire to make sure they get GC'ed.  I probably wrote it myself and was terrified about hte implications of holding onto workspaces in closures.

Comment (Pilchie) [inline src/EditorFeatures/Core/Implementation/Preview/PreviewFactoryService.cs]: I believe @heejaechang did a hotfix for the fact that we were leaking workspaces from previews.

**Comment by Pilchie** [comment]
Do we want this for `dev15-rc2`?

**Comment by CyrusNajmabadi** [comment]
Dunno.  We've only heard about this from @kuhlenh Right?

**Pull request merged**